vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+    The **`fill`** aesthetic is also set to **`rep_name`**, which maps each bar to a unique color based on the **`rep_name`** variable.
+    The function also adds labels to the x and y axes and applies the **`theme_light()`** theme to the plot.
+    Finally, the function is exported with **`#' @export`**, which makes it available to other packages and scripts that import this one.
+
+### Call the helper function in the desired `rhino` module
+
+-   Now let us go back to the `app\view\datatable.R` file.
+    Here we need to call the `generate_barchart()` function we just created along with the required dependencies to render it on UI as a plotly bar chart.
+
+-   So, the code begins with
+
+    ```         
+    box::use(
+      shiny[moduleServer, NS, uiOutput, renderUI, reactive, div],
+      shiny.fluent[fluentPage, Text, DetailsList, Toggle.shinyInput],
+      tibble[tibble],
+      dplyr[filter],
+      plotly[plotlyOutput, renderPlotly],
+    )
+
+    box::use(
+      app/logic/barchart[generate_barchart],
+    )
+    ```
+
+    Here, we have called the `plotly` library, and also the function that we defined in the `app/logic` folder.
+
+-   Now, we will add one more `div` in the ui section after the datatable to make room for the bar chart
+
+````         
+```         
+ div(
+      class = "ms-depth-64",
+      style = "height: 500px; overflow: auto;",
+      plotlyOutput(ns("barchart"))
+      )
+```
+
+`plotlyOutput` will help us to render the graph
+````
+
+-   Similarly, in the server section, we will need to add some code to ask the backend of the application for rendering the barchart
+
+    ```         
+     output$barchart <- renderPlotly({
+            generate_barchart(filtered_deals())
+        })
+    ```
+
+Once ready, on running the application, we should see the plotly bar chart rendering successfully and also responding to our interactions with the toggle input button.
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_4.png")
+```
+
+## 6. Some Improvements for better User Experience
+
+### Keep the data table and bar chart on the same row
+
+This can be done by wrapping the `div`s into a parent `div` and using the **`container-fluid`** class to ensure that the **`div`**s fill the screen horizontally.
+
+```         
+div(
+          class = "container-fluid",
+          style = "margin-left: 0px; margin-right: 0px;",
+          div(
+              class = "ms-depth-8",
+              style = "height: 500px; overflow: auto; width: 50%; display: inline-block;",
+              uiOutput(ns("datatable"))
+          ),
+          div(
+              class = "ms-depth-8",
+              style = "height: 500px; overflow: auto; width: 50%; display: inline-block;",
+              plotlyOutput(ns("barchart"))
+          ),
+          style = "display: flex; flex-wrap: wrap;"
+        )
+```
```

Comment:
Avoid inline styling.

Reply:
added comment to redirect user to custom styling in rhino documentation